File: xamarin/Samples/Showcase-Android/Activities/CommentAddActivity.cs
using Android.App;
using Android.OS;
using Android.Widget;
using Com.Liferay.Mobile.Screens.Comment;
using Com.Liferay.Mobile.Screens.Comment.Add;

namespace ShowcaseAndroid
{
    [Activity]
    public class CommentAddActivity : Activity, ICommentAddListener
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.CommentAddView);

            CommentAddScreenlet commentAddScreenlet =
                (CommentAddScreenlet) FindViewById(Resource.Id.comment_add_screenlet);
            commentAddScreenlet.Listener = this;
        }

        /* ICommentAddListener */

        public void Error(Java.Lang.Exception p0, string p1)
        {
            System.Diagnostics.Debug.WriteLine($"Comment add error: {p0.Message}");
        }

        public void OnAddCommentSuccess(CommentEntry p0)
        {
            Toast.MakeText(this, "Comment add success: " + p0.CommentId, ToastLength.Short).Show();
        }
    }
}

File: android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/CommentAddActivity.java
package com.liferay.mobile.screens.testapp;

import android.os.Bundle;
import com.liferay.mobile.screens.comment.CommentEntry;
import com.liferay.mobile.screens.comment.add.CommentAddListener;
import com.liferay.mobile.screens.comment.add.CommentAddScreenlet;

/**
 * @author Sarai Díaz García
 */
public class CommentAddActivity extends ThemeActivity implements CommentAddListener {

    private CommentAddScreenlet commentAddScreenlet;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.comment_add);

        commentAddScreenlet = findViewById(R.id.comment_add_screenlet);
        commentAddScreenlet.setListener(this);
    }

    @Override
    public void onAddCommentSuccess(CommentEntry commentEntry) {
        info(getString(R.string.comment_added_success));
    }

    @Override
    public void error(Exception e, String userAction) {
        error(getString(R.string.comment_error) + " " + userAction, e);
    }
}
